plugins/kubernetes-cluster/src/components/ApiResources/ApiResources.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useApiResources } from './useApiResources';
+import React from 'react';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { Table, TableColumn } from '@backstage/core-components';
+import { IAPIGroup } from '@kubernetes-models/apimachinery/apis/meta/v1';
+import { useKubernetesClusterError } from '../KubernetesClusterErrorContext/KubernetesClusterErrorContext';
+import { makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles(theme => ({
+  empty: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    justifyContent: 'center',
+  },
+}));
+
+const defaultColumns: TableColumn<IAPIGroup>[] = [
+  {
+    title: 'Name',
+    highlight: true,
+    render: (apiGroup: IAPIGroup) => {
+      return apiGroup.name;
+    },
+  },
+  {
+    title: 'Preferred Version',
+    highlight: true,
+    render: (apiGroup: IAPIGroup) => {
+      return apiGroup.preferredVersion?.groupVersion;
+    },
+  },
+];
+
+export const ApiResources = () => {
+  const classes = useStyles();
+  const { entity } = useEntity();
+  const { value, error, loading } = useApiResources({
+    clusterName: entity.metadata.name,
+  });
+  const { setError } = useKubernetesClusterError();
+
+  if (error) {
+    setError(error.message);
+  }
```

Comment:
this block should be moved in a `useEffect`:

```suggestion
  useEffect(() => {
    if(error) {
      setError(error.message);
    }
  }, [error]);
```

same as in other components. You could also move the `useEffect` inside `useApiResources` so that you remove some of the existing logic from the components

Reply:
fixed!